stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -184,6 +184,9 @@ Random.seed!(1)
         @test (r = Matrix(D)' * vv  ; mul!(vvv, adjoint(D), vv) ≈ r ≈ vvv)
         @test (r = transpose(Matrix(D)) * vv ; mul!(vvv, transpose(D), vv) ≈ r ≈ vvv)
 
+        UUU = similar(UU)
+        @test (r = UU * Matrix(D)   ; mul!(UUU, UU, D) ≈ r ≈ UUU)
```

Comment:
The following tests would need to be added:
```
@test (r = UU * Matrix(D)' ; mul!(UUU, UU, adjoint(D)) ≈ r ≈ UUU)
@test (r = UU * transpose(Matrix(D)); mul!(UUU, UU, transpose(D)) ≈ r ≈ UUU)
```

Reply:
Although this mimics the previous tests, it's better to split the like instead of using `;` as it improves the readability.